### ISSUE TITLE
Bug report: requiring the node module is incredibly slow

### ISSUE THREAD

**Issue by Prinzhorn** [gchq/CyberChef#1016]
**Describe the bug**
`require('cyberchef')` takes roughly 3s to 4s. Tested on both Node.js 10 and 12. Might be much worse without a fast SSD.

**To Reproduce**

```js
console.time('require');
require('cyberchef');
console.timeEnd('require');
```

**Expected behaviour**
A couple of ms

**Desktop (if relevant, please complete the following information):**
 - OS: Ubuntu 19.10
 - Browser: Node.js
 - CyberChef version: 9.20.3

**Screenshots**

![Selection_608](https://user-images.githubusercontent.com/679144/79436528-4aea8f80-7fd1-11ea-98dc-4b8c52ec0bce.png)

**Additional context**

3s might be _somewhat_ acceptable in a long-running server process, but unfortunately it's beyond acceptable in an Electron app where startup time without cyberchef is otherwise instant.

It appears that a lot of time is spent by "esm.js". I tried compiling a wrapper script using [ncc](https://github.com/zeit/ncc/) but it fails. It seems like it tries to load some of the web stuff, even if I explicitly `require('cyberchef/src/node/index.mjs')`

Using `import` with `--experimental-modules` doesn't make it fast at all. Edit: correction, explicitly doing `import chef from "cyberchef/src/node/index.mjs";` brings it down to like 2s.

Would it be possible to compile a single bundle for node just like it's done for the web version?

**Comment by Prinzhorn** [comment]
I dug into this just a tiny bit (and was looking at https://github.com/gchq/CyberChef/pull/284) and I'm wondering if there are plans to move to a more standard approach to modules? What I mean by that is that this looks like a mix of Grunt, webpack and esm (which seems unmaintained and unnecessary?). I don't even know where to start making sense of everything.

I understand that this was historically a Browser-only application and has then been refactored to also support Node.js. Which explains where this is coming from. But when I'm consuming a package in Node.js, I expect it to use the built-in module system. Right now even the Node.js build relies on webpack and then _at runtime_ it uses a module loader (esm.js), written in JavaScript.

I will look into this again in the future, as I'm at least looking for a workaround to compile a Node.js version that loads in the ms range, preferably single-digit. I will report back once I'm happy with the results.

**Comment by d98762625** [comment]
Hello, thanks for your issue. This had prompted me to have a dig through NodeJS docs around ESM/CJS interoperability and hopefully the PR above will help improve things.

As you say CyberChef was written for the web. It's written entirely in ES Modules, which are historically incompatible with CommonJS modules, Node.js's module ecosystem. 

Later versions of node are [improving interoperability between ESM and CJS](https://nodejs.org/docs/latest-v12.x/api/esm.html#esm_ecmascript_modules), but its not perfect. Using the `esm` package allows us to export the CyberChef Node API so that it is fully compatible with CJS by default, but as you've raised, that gives a significant performance hit. Thanks for your flame graph by the way, I've tried out something similar in the above PR. We need to use ESM here because [the only "native" way to load an ES module from a CJS module is to use `import()`](https://nodejs.org/docs/latest-v12.x/api/esm.html#esm_interoperability_with_commonjs), which would force every CSJ consumer of CyberChef to load it asynchronously. 

Take a look at the above PR, see if it helps your cause. I would recommend trying the `import` path, as I've used [package exports](https://nodejs.org/docs/latest-v12.x/api/esm.html#esm_package_entry_points) to define different entrypoints for `import` and `require`. This comes with the caveat that you'll have to add some flags (documented in the PR) at runtime.

I'd be interested to know if this helps with your performance issues.
